File: frontend/src/api/knowledge.ts
import client from './client';

export interface Document {
  id: string;
  filename: string;
  file_type: string;
  file_size: number;
  status: string;
  char_count: number;
  chunk_count: number;
  uploaded_by: string;
  error_message?: string;
  created_at?: string;
  updated_at?: string;
}

export async function getDocuments(params: {
  page?: number;
  size?: number;
  status?: string;
  search?: string;
}): Promise<{ items: Document[]; total: number }> {
  const res = await client.get('/api/knowledge/documents', { params });
  return res.data;
}

export async function uploadDocument(file: File): Promise<Document> {
  const formData = new FormData();
  formData.append('file', file);
  const res = await client.post('/api/knowledge/documents/upload', formData, {
    headers: { 'Content-Type': 'multipart/form-data' },
  });
  return res.data;
}

export async function deleteDocument(docId: string): Promise<void> {
  await client.delete(`/api/knowledge/documents/${docId}`);
}

export async function getDocumentStatus(docId: string): Promise<{
  id: string;
  status: string;
  char_count: number;
  chunk_count: number;
}> {
  const res = await client.get(`/api/knowledge/documents/${docId}/status`);
  return res.data;
}

export async function getStats(): Promise<{
  document_count: number;
  total_chunks: number;
  total_chars: number;
}> {
  const res = await client.get('/api/knowledge/stats');
  return res.data;
}

File: backend/app/schemas/knowledge.py
"""知识库相关的 Pydantic 模型"""
from pydantic import BaseModel, Field


class DocumentResponse(BaseModel):
    id: str
    filename: str
    file_type: str
    file_size: int
    status: str
    char_count: int
    chunk_count: int
    uploaded_by: str
    error_message: str | None = None
    created_at: str | None = None
    updated_at: str | None = None


class DocumentListResponse(BaseModel):
    items: list[DocumentResponse]
    total: int


class DocumentStatusResponse(BaseModel):
    id: str
    status: str
    char_count: int
    chunk_count: int


class KnowledgeStatsResponse(BaseModel):
    document_count: int
    total_chunks: int
    total_chars: int
